admin_pages/messages/Messages_Admin_Page.core.php

```diff
@@ -2973,18 +2947,45 @@ protected function _insert_or_update_message_template($new = false)
             $override = true;
         }
 
-        if (empty($query_args)) {
-            $query_args = [
+        $query_args = ! empty($query_args)
+            ? $query_args
+            : [
                 'id'      => $GRP_ID,
                 'context' => $context_slug,
                 'action'  => 'edit_message_template',
             ];
-        }
 
         $this->_redirect_after_action($success, $item_desc, $action_desc, $query_args, $override);
     }
 
 
+    /**
+     * recursively runs wp_kses() on message template content in a model safe manner
+     *
+     * @param array|string $content
+     * @return array|string
+     * @since   $VID:$
+     */
+    private function sanitizeMessageTemplateContent($content)
+    {
+        if (is_array($content)) {
+            foreach ($content as $key => $value) {
+                $content[ $key ] = $this->sanitizeMessageTemplateContent($value);
+            }
+            return $content;
+        }
+        // remove slashes so wp_kses() works properly
+        // wp_kses_stripslashes() only removes slashes from double-quotes,
+        // so attributes using single quotes always appear invalid.
+        $content = stripslashes($content);
+        $content = wp_kses($content, wp_kses_allowed_html('post'));
+        // But currently the models expect slashed data, so after wp_kses()
+        // runs we need to re-slash the data. Sheesh.
+        // See https://events.codebasehq.com/projects/event-espresso/tickets/11211#update-47321587
+        return addslashes($content);
```

Comment:
🤨 😅

Reply:
pretty sure that's Darren's comments and code